sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -369,28 +373,120 @@ public static String stringJoin(String delimiter, List<String> values) {
+     * <p>
+     * If {@code uppercase} is true {@code ABCDEF} will be used, if false {@code abcdef} will be used. No matter the
+     * value of {@code uppercase} characters {@code 0123456789} remain consistent.
+     * <p>
+     * If {@code bytes} is null, null will be returned. If {@code bytes} was an empty array an empty string is
+     * returned.
+     *
+     * @param bytes The byte array to convert into a hex string.
+     * @param uppercase Whether {@code abcdef} or {@code ABCDEF} are used for the hex string.
+     * @return A hex string representing the {@code bytes} that were passed, or null if {@code bytes} were null.
+     */
+    public static String bytesToHexString(byte[] bytes, boolean uppercase) {
+        if (bytes == null) {
+            return null;
+        }
+
+        if (bytes.length == 0) {
+            return "";
+        }
+
+        // Hex uses 4 bits, converting a byte to hex will double its size.
+        char[] hexString = new char[bytes.length * 2];
+        char[] hexCharacters = uppercase ? UPPERCASE_HEX : LOWERCASE_HEX;
+
+        for (int i = 0; i < bytes.length; i++) {
+            // Convert the byte into an integer, masking all but the last 8 bits (the byte).
+            int b = bytes[i] & 0xFF;
+
+            // Shift 4 times to the right to get the leading 4 bits and get the corresponding hex character.
+            hexString[i * 2] = hexCharacters[b >>> 4];
+
+            // Mask all but the last 4 bits and get the corresponding hex character.
+            hexString[i * 2 + 1] = hexCharacters[b & 0x0F];
         }
+
+        return new String(hexString);
+    }
+
+    /**
+     * Extracts the size from a {@code Content-Range} header.
+     * <p>
+     * The {@code Content-Range} header can take the following forms:
+     *
+     * <ul>
+     * <li>{@code <unit> <start>-<end>/<size>}</li>
+     * <li>{@code <unit> <start>-<end>/}&#42;</li>
+     * <li>{@code <unit> }&#42;{@code /<size>}</li>
+     * </ul>
+     *
+     * If the {@code <size>} is represented by &#42; this method will return -1.
+     * <p>
+     * If {@code contentRange} is null a {@link NullPointerException} will be thrown, if it doesn't contain a size
+     * segment ({@code /<size>} or /&#42;) -2 will be returned.
+     *
+     * @param contentRange The {@code Content-Range} header to extract the size from.
+     * @return The size contained in the {@code Content-Range}, -1 if the size was &#42;, -2 if there was no size
+     * segment.
+     * @throws NullPointerException If {@code contentRange} is null.
+     * @throws NumberFormatException If the {@code <size>} segment of the {@code contentRange} isn't a valid number.
+     */
+    public static long extractSizeFromContentRange(String contentRange) {
+        Objects.requireNonNull(contentRange, "Cannot extract length from null 'contentRange'.");
+        int index = contentRange.indexOf('/');
+
+        if (index == -1) {
+            // No size segment.
+            return -2;
```

Comment:
I'd prefer it to log and throw - it's an invalid Content-Range header and there is no recovery from it